core/sql/sqlcomp/CmpSeabaseDDLcommon.cpp

```diff
@@ -5322,19 +5395,74 @@ short CmpSeabaseDDL::updateObjectRedefTime(
   NAString quotedObjName;
   ToQuotedString(quotedObjName, NAString(objName), FALSE);
 
-  str_sprintf(buf, "update %s.\"%s\".%s set redef_time = %Ld where catalog_name = '%s' and schema_name = '%s' and object_name = '%s' and object_type = '%s' ",
-              getSystemCatalog(), SEABASE_MD_SCHEMA, SEABASE_OBJECTS,
-              redefTime,
-              catName.data(), quotedSchName.data(), quotedObjName.data(),
-              objType);
+  Int64 flags = 0;
+  if (((CmpCommon::getDefault(TRAF_STORE_OBJECT_DESC) == DF_ON) ||
+       (force)) &&
+      (objUID > 0) &&
+      (NOT isSeabaseReservedSchema(catName, schName)) &&
+      ((strcmp(objType, COM_BASE_TABLE_OBJECT_LIT) == 0) ||
+       (strcmp(objType, COM_VIEW_OBJECT_LIT) == 0)))
+    {
+      Int32 ctlFlags = GEN_PACKED_DESC | GET_SNAPSHOTS;
+      Int32 packedDescLen = 0;
+      TrafDesc * ds = 
+        getSeabaseUserTableDesc(catName, schName, objName,
+                                (strcmp(objType, COM_BASE_TABLE_OBJECT_LIT) == 0
+                                 ? COM_BASE_TABLE_OBJECT : COM_VIEW_OBJECT),
+                                FALSE, ctlFlags, packedDescLen);
+      if (! ds)
+        {
+          processReturn();
+          return -1;
+        }
+
+      cliRC = updateTextTableWithBinaryData
```

Comment:
We update the stored descriptor here if TRAF_STORE_OBJECT_DESC is set to ON. What about the case when a table already has a stored object descriptor and TRAF_STORE_OBJECT_DESC is OFF? Should we at least delete the outdated stored descriptor? Sorry, I didn't check the details, probably this situation can never happen?